src/components/ListItem.jsx

```diff
@@ -1,5 +1,76 @@
+import { useEffect, useState } from 'react';
+import { useToggle } from '@uidotdev/usehooks';
+import { Toggle } from './Toggle.jsx';
 import './ListItem.css';
+import { updateItem } from '../api/firebase.js';
+// import { getFutureDate} from '../utils/dates.js'
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({
+	name,
+	listPath,
+	itemId,
+	totalPurchases,
+	dateLastPurchased,
+}) {
+	const [purchased, setPurchased] = useToggle(false);
+	const [isDisabled, setIsDisabled] = useState(false);
+
+	useEffect(() => {
+		if (dateLastPurchased) {
+			const checkExpiration = () => {
+				const expirationDate = new Date(
+					dateLastPurchased.toMillis() + 24 * 60 * 60 * 1000,
+				);
+				const currentDate = new Date();
+				if (currentDate > expirationDate) {
+					setPurchased(false);
+					setIsDisabled(false);
+				} else {
+					setPurchased(true);
+					setIsDisabled(true);
+				}
+			};
+			checkExpiration();
+			const refreshInterval = setInterval(checkExpiration, 60000);
+			return () => clearInterval(refreshInterval);
+		}
+	}, [dateLastPurchased]);
+
+	const handleToggle = async () => {
+		const isPurchasing = !purchased;
+		setPurchased();
+
+		if (isPurchasing) {
+			try {
+				await updateItem(listPath, {
+					itemId,
+					totalPurchases: totalPurchases + 1,
+					dateLastPurchased: new Date(),
+				});
+				console.log(`${name} updated successfully`);
+				alert(`${name} is purchased successfully`);
+				setIsDisabled(true);
+			} catch (error) {
+				console.error('Error updating item:', error);
+				setPurchased(false);
+			}
+		}
+	};
+
+	return (
+		<div>
```

Comment:
I'm noticing that there are a lot of divs in this component, which is cluttering up the DOM structure a bit (sometimes this is called ["div soup"](https://www.reddit.com/r/css/comments/146c56y/what_is_div_soup/)).

Generally, I try to reduce the number of DOM elements as much as possible, leaving only as many as I need for semantic markup & for styling. In this case, I think you could eliminate this wrapper div and just return the li.

(If you ever run into issues because your component needs to return a single element, you can wrap the other elements in a [React Fragment](https://react.dev/reference/react/Fragment), which acts as a wrapper but doesn't actually render anything in the DOM.)